### PR TITLE
Added pundit authorization

### PR DESCRIPTION
**Pull request by dgarcia13-99**
Added pundit authorization to comments, photos, follow_requests, and users.

### CODE REVIEW

File: app/controllers/application_controller.rb
Comment (hatcheta1) [inline app/controllers/application_controller.rb:1]: nice job securing by default!

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
-
+  before_action {authorize(@comment || Comment)}
```
Comment (hatcheta1) [inline app/controllers/comments_controller.rb:3]: great use of ```before_action```!

File: app/controllers/users_controller.rb
Comment (hatcheta1) [inline app/controllers/users_controller.rb:1]: Screenshot 2024-11-01 at 3 52 23 PM
got this error when I loaded the live app preview

**Comment by armstrca** [comment]
Alice seems to be able to edit and delete anyone else's photos and comments still. The actions shouldn't work and the icons shouldn't show up.

Screenshot 2024-11-04 120147
Screenshot 2024-11-04 120122

**Review by armstrca — CHANGES_REQUESTED** [review]
Make sure to test things out in the live preview before committing/making PRs.

Current_user should be able to see a private user's profile page, but *not* the private user's photos.

File: app/controllers/users_controller.rb
Comment (armstrca) [reply app/controllers/users_controller.rb:1]: Same.

Comment (dgarcia13-99) [reply app/controllers/users_controller.rb:1]: Thank you for the feedback! I had a syntax error, but it should be working now!

File: app/controllers/photos_controller.rb
Comment (armstrca) [inline app/controllers/photos_controller.rb:1]: Screenshot 2024-11-05 115913

File: app/controllers/comments_controller.rb
Comment (armstrca) [inline app/controllers/comments_controller.rb:1]: Screenshot 2024-11-05 120011

File: app/views/photos/show.html.erb
Comment (armstrca) [inline app/views/photos/show.html.erb:1]: Put some handling in here for cases in which `photo.owner` is private and `current_user` is not a follower of `photo.owner`

**Review by armstrca — CHANGES_REQUESTED** [review]

**Review by armstrca — APPROVED** [review]
boom

**Pull request merged**